### PR TITLE
Bug 2079468: Enhance the waitForIngressControllerCondition for better CI results

### PR DESCRIPTION
**Pull request by candita**
Due to numerous recent failures in matching expected to current conditions, add some logging to provide better CI results.  https://bugzilla.redhat.com/show_bug.cgi?id=2079468

### CODE REVIEW

File: test/e2e/operator_test.go [hunk 1]
```diff
@@ -2942,17 +2942,34 @@ func waitForRouteIngressConditions(t *testing.T, cl client.Client, routeName typ
...
-		expected := operatorConditionMap(conditions...)
-		current := operatorConditionMap(ic.Status.Conditions...)
+		expected = operatorConditionMap(conditions...)
+		current = operatorConditionMap(ic.Status.Conditions...)
+
 		return conditionsMatchExpected(expected, current), nil
 	})
+
+	if err != nil {
+		t.Errorf("Expected conditions: %v\n Current conditions: %v", expected, current)
```
Comment (frobware) [inline test/e2e/operator_test.go:2970]: It would be easier to read and digest if we emitted diff output here.

Comment (candita) [reply test/e2e/operator_test.go:2970]: Hi Andy - this is a quick fix for extra knowledge and I wasn't planning on putting any more time into it.  Do you mind if I leave as-is, and someone else can update later if they want?

Comment (frobware) [reply test/e2e/operator_test.go:2970]: OK for me.

**Comment by candita** [comment]
> reason: 'LeaderElection' ci-op-r4rdtm6y-6da5c-csnhs-master-0 stopped leading\nE0427 16:11:28.229071       1 network_controller.go:97] leader election lost\n

/retest

**Comment by candita** [comment]
ERROR	operator.main	cobra/command.go:860	error starting	{"error": "failed to wait for canary_controller caches to sync: timed out waiting for cache to be synced"}

/test e2e-gcp-serial

**Pull request merged**